expression/schema.go

```diff
@@ -102,9 +129,34 @@ func (s *Schema) Append(col *Column) {
 	s.Columns = append(s.Columns, col)
 }
 
+// SetUniqueKeys will set the value of Schema.Keys.
+func (s *Schema) SetUniqueKeys(keys []KeyInfo) {
+	s.Keys = keys
+}
+
+// GetColumnsIndices will return a slice which contains the position of each column in schema,
+// and a boolean which tell you whether all columns are found.
+func (s Schema) GetColumnsIndices(cols []*Column) (ret []int, allFound bool) {
+	allFound = true
+	for _, col := range cols {
+		pos := s.GetColumnIndex(col)
+		if pos != -1 {
+			ret = append(ret, pos)
+		} else {
+			ret = append(ret, -1)
```

Comment:
if pos == 1, return nil directly, then you can check empty by checking if the ret is nil